test/functional/tests/security/test_security_user.py

```diff
@@ -354,10 +354,12 @@ def test_user_cli():
 
     with TestRun.step("Try to zero metadata with 'sudo'."):
         try:
-            run_as_other_user(cli.zero_metadata_cmd(str(cache_dev)),
-                              user_name, True)
+            output = run_as_other_user(cli.zero_metadata_cmd(str(cache_dev.path)), user_name, True)
+            if output.exit_code == 0:
```

Comment:
This `if` is actually unnecessary. We expect `run_as_other_user()` to raise an exception, so if that does not happen we can just error out.

Reply:
I may not understand ^^ this "if statement " and next line is for the error case = " if that does not happen we can just error out".
This is exactly what you've proposed I guess. And I think (and hope) the messages are helpful for understanding tests intentions and what is expected here.